shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -175,6 +176,141 @@ abstract class JSBackend(allowUnresolvedSymbols: Bool) {
     case _ => throw CodeGenError(s"ill-formed application ${inspect(term)}")
   }
 
+  private def createASTCall(tp: Str, args: Ls[Term]): App =
+    App(Var(tp), Tup(args.map(a => N -> Fld(FldFlags.empty, a))))
+
+  private def desugarQuotedBranch(branch: CaseBranches)(
+    implicit scope: Scope, isQuoted: Bool, freeVars: MutSet[Str]
+  ): Either[Term, CaseBranches] = branch match {
+    case Case(pat, body, rest) =>
+      val dp = desugarQuote(pat)
+      val db = desugarQuote(body)
+      desugarQuotedBranch(rest) match {
+        case L(t) => L(createASTCall("Case", dp :: db :: t :: Nil))
+        case R(b) => dp match {
+          case dp: SimpleTerm => R(Case(dp, db, b))
+          case _ => die
+        }
+      }
+    case Wildcard(body) =>
+      if (isQuoted) L(createASTCall("Wildcard", desugarQuote(body) :: Nil)) else R(Wildcard(desugarQuote(body)))
+    case NoCases => if (isQuoted) L(createASTCall("NoCases", Nil)) else R(NoCases)
+  }
+
+  private def toJSOperator(op: Str) = op match {
+    case "+." => "+"
+    case "-." => "-"
+    case "*." => "*"
+    case _ => op
+  }
+
+  private def desugarQuote(term: Term)(implicit scope: Scope, isQuoted: Bool, freeVars: MutSet[Str]): Term = term match {
```

Comment:
Document this is a temporary hack due to lack of runtime support.

Reply:
I just noticed that `desugarQuote` should not be a temporary hack. For example, we need to desugar `Quote(IntLit(42))` into JS code `IntLit(42)`, where `IntLit` is an AST constructor function. I've documented it in the comment.